service/src/main/java/org/cbioportal/service/impl/GenericAssayServiceImpl.java

```diff
@@ -74,14 +75,19 @@
                     metaResults.add(new GenericAssayMeta(meta.getStableId()));
                 }
             } else {
-                // get additional properties for each meta data
+                List<GenericAssayAdditionalProperty> additionalProperties = genericAssayRepository.getGenericAssayAdditionalproperties(distinctStableIds);
                 for (GenericAssayMeta meta : metaData) {
-                    int geneticEntityId = genericAssayRepository.getGeneticEntityIdByStableId(meta.getStableId());
+                    String stableId = meta.getStableId();
                     HashMap<String, String> map = new HashMap<>();
-                    for (HashMap<String, String> data : genericAssayRepository.getGenericAssayMetaPropertiesMap(geneticEntityId)) {
-                        map.put(data.get("key"), data.get("value"));
+                    List<GenericAssayAdditionalProperty> filteredAdditionalProperties = additionalProperties
+                            .stream()
+                            .filter(additionalProperty -> additionalProperty.getStableId().equals(stableId))
+                            .collect(Collectors.toList());
+                    for (GenericAssayAdditionalProperty additionalProperty : filteredAdditionalProperties) {
+                        map.put(additionalProperty.getName(), additionalProperty.getValue());
                     }
-                    metaResults.add(new GenericAssayMeta(meta.getEntityType(), meta.getStableId(), map));
+                    meta.setGenericEntityMetaProperties(map);
+                    metaResults.add(meta);
```

Comment:
Main change is at here: we call function `getGenericAssayAdditionalproperties` once instead of calling N times `getGeneticEntityIdByStableId` and `getGenericAssayMetaPropertiesMap` (N is the number of stable ids).